net/relayd/src/opnsense/mvc/app/controllers/OPNsense/Relayd/Api/ServiceController.php

```diff
@@ -67,6 +87,49 @@ public function configtestAction()
         }
     }
 
+    /**
+     * reconfigure relayd
+     * @return array
+     */
+    public function reconfigureAction()
+    {
+        if ($this->request->isPost()) {
+            $this->sessionClose();
```

Comment:
If you don't close the session, lock() is built-in. For serialisation purposes when you really want this lock to go into the background for your own plugin, it should lock first, then close the session.

Reply:
What do mean with "built-in" and "background"?

Reply:
If the session isn't closed, it creates a global type of lock in the API.

Reply:
Does it mean that not closing the session the whole API is locked?

Reply:
sessionClose() simply calls session_write_close() that closes the session and unlocks the session data.
My intention is to serialize the configuration creation/reload of the service. Therefor the lock is in the scope of the plugin (more eaxactly the HTTP server thread) and not the session.

Reply:
For the sake of shipping version 2.0 of this plugin please move the session close after the lock acquire. I can stop explaining how this stuff works if you think this is not necessary for you but we will have to do do the review and code improvement either way.